<launch>

    <!-- Do we want an output? if so, add view:=true to launch command-->
    <arg name="view" default="false" />

    <!-- Launch person following robot image analysis node-->
    <node name="image_analyzer" pkg="pfr_package" type="prf_image_evaluation.py" output="screen" />

    <!-- Launch the USB camera -->
    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen">
        <param name="video_device" value="/dev/video0" />
        <!-- param name="video_device" value="/dev/video2" / -->
        <param name="image_width" value="640" />
        <param name="image_height" value="480" />
        <param name="pixel_format" value="yuyv" />
        <param name="color_format" value="yuv422p" />
        <param name="camera_frame_id" value="usb_cam" />
        <param name="io_method" value="mmap" />
    </node>

    <!-- Open a window to view the image if view arg is true -->
    <node if="$(arg view)" name="image_view" pkg="image_view" type="image_view" respawn="false">
        <remap from="image" to="/usb_cam/image_raw" />
        <param name="autosize" value="true" />
    </node>

    <!-- Console launch prefix -->
    <arg name="launch_prefix" default=""/>
    <arg name="image" default="/camera/rgb/image_raw" />

    <!-- Config and weights folder. -->
    <arg name="yolo_weights_path"          default="$(find darknet_ros)/yolo_network_config/weights"/>
    <arg name="yolo_config_path"           default="$(find darknet_ros)/yolo_network_config/cfg"/>

    <!-- ROS and network parameter files -->
    <arg name="ros_param_file"             default="$(find darknet_ros)/config/ros.yaml"/>
    <arg name="network_param_file"         default="$(find darknet_ros)/config/yolov2-tiny.yaml"/>

    <!-- Load parameters -->
    <rosparam command="load" ns="darknet_ros" file="$(arg ros_param_file)"/>
    <rosparam command="load" ns="darknet_ros" file="$(arg network_param_file)"/>

    <!-- Start darknet and ros wrapper -->
    <node pkg="darknet_ros" type="darknet_ros" name="darknet_ros" output="log" launch-prefix="$(arg launch_prefix)">
      <param name="weights_path"          value="$(arg yolo_weights_path)" />
      <param name="config_path"           value="$(arg yolo_config_path)" />
      <remap from="camera/rgb/image_raw"  to="$(arg image)" />
    </node>
    
</launch>
